src/serveractiveobjectmap.h

```diff
@@ -0,0 +1,81 @@
+/*
+Minetest
+Copyright (C) 2010-2017 numZero, Lobachevsky Vitaly <numzer0@yandex.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+#include <set>
+#include <unordered_map>
+#include <vector>
+#include "irr_v3d.h"
+#include "irr_aabb3d.h"
+
+class ServerActiveObject;
+
+namespace std
+{
+	template<> struct hash<v3s16> {
+		typedef v3s16 argument_type;
+		typedef std::size_t result_type;
+		result_type operator() (const argument_type &s) const noexcept
+		{
+			return std::hash<u64>()(
```

Comment:
if you implement this hash it should be elsewhere because we need it in many placed to use unordered_maps